internal/pkg/cli/app_package.go

```diff
@@ -118,19 +132,39 @@ func (o *packageAppOpts) Execute() error {
 	}
 
 	if o.OutputDir != "" {
-		if err := o.setFileWriters(); err != nil {
+		if err := o.setAppFileWriters(); err != nil {
 			return err
 		}
 	}
 
-	templates, err := o.getTemplates(env)
+	appTemplates, err := o.getAppTemplates(env)
 	if err != nil {
 		return err
 	}
-	if _, err = o.stackWriter.Write([]byte(templates.stack)); err != nil {
+	if _, err = o.stackWriter.Write([]byte(appTemplates.stack)); err != nil {
+		return err
+	}
+	if _, err = o.paramsWriter.Write([]byte(appTemplates.configuration)); err != nil {
 		return err
 	}
-	_, err = o.paramsWriter.Write([]byte(templates.configuration))
+
+	addonsTemplate, err := o.getAddonsTemplate()
+	// return nil if addons dir doesn't exist.
+	notExistErr := &workspace.ErrAddonsDirNotExist{AppName: o.AppName}
+	if errors.As(err, &notExistErr) {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("retrieve addons template: %w", err)
+	}
+
+	if o.OutputDir != "" {
```

Comment:
Maybe a comment here saying that why we're doing this, and maybe we should warn folks that there's infrastructure, but we're not showing it?